lib/holidays/date_calculator/lunar_date.rb

```diff
@@ -340,18 +340,173 @@ def to_s
+        [354, 1, 1, 2, 1, 1, 2, 1, 2, 2, 2, 1, 2].freeze,
+        [354, 2, 1, 1, 2, 1, 1, 2, 1, 2, 2, 1, 2].freeze,
+        [384, 2, 2, 1, 1, 4, 1, 2, 1, 2, 1, 2, 2].freeze,
+        [354, 2, 1, 2, 1, 2, 1, 1, 2, 1, 2, 1, 2].freeze,
+        [354, 2, 1, 2, 2, 1, 2, 1, 1, 2, 1, 2, 1].freeze,
+        [384, 2, 1, 2, 4, 2, 1, 2, 1, 2, 1, 2, 1].freeze,
+        [355, 2, 1, 2, 1, 2, 2, 1, 2, 1, 2, 1, 2].freeze,
+        [384, 1, 2, 1, 2, 1, 2, 1, 2, 4, 2, 1, 2].freeze,
+        [354, 1, 2, 1, 1, 2, 1, 2, 2, 2, 1, 2, 1].freeze,
+        [355, 2, 1, 2, 1, 1, 2, 1, 2, 2, 1, 2, 2].freeze,
+        [384, 1, 2, 1, 2, 1, 5, 1, 2, 1, 2, 2, 2].freeze,
+        [354, 1, 2, 1, 2, 1, 1, 2, 1, 2, 1, 2, 2].freeze,
+        [354, 2, 1, 2, 1, 2, 1, 1, 2, 1, 1, 2, 2].freeze,
+        [384, 1, 2, 2, 4, 2, 1, 1, 2, 1, 2, 1, 2].freeze,
+        [354, 1, 2, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1].freeze,
+        [355, 2, 1, 2, 1, 2, 2, 1, 2, 1, 2, 1, 2].freeze,
+        [384, 1, 4, 1, 2, 2, 1, 2, 2, 1, 2, 1, 2].freeze,
+        [354, 1, 2, 1, 1, 2, 1, 2, 2, 1, 2, 2, 1].freeze,
+        [384, 2, 1, 2, 1, 1, 4, 2, 1, 2, 2, 2, 1].freeze,
+        [354, 2, 1, 2, 1, 1, 2, 1, 1, 2, 2, 2, 1].freeze,
+        [354, 2, 2, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1].freeze,
+        [384, 2, 2, 2, 1, 4, 1, 2, 1, 1, 2, 2, 1].freeze,
+        [355, 2, 2, 1, 2, 1, 2, 1, 2, 1, 1, 2, 2].freeze,
+        [354, 1, 2, 1, 2, 2, 1, 2, 1, 2, 1, 2, 1].freeze,
+        [384, 1, 2, 4, 2, 1, 2, 2, 1, 2, 1, 2, 1].freeze,
+        [355, 2, 1, 1, 2, 1, 2, 2, 1, 2, 2, 1, 2].freeze,
+        [384, 1, 2, 1, 1, 2, 1, 2, 1, 2, 2, 4, 2].freeze,
+        [354, 1, 2, 1, 1, 2, 1, 2, 1, 2, 2, 1, 2].freeze,
+        [354, 2, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 2].freeze,
+        [384, 2, 2, 1, 2, 1, 5, 1, 1, 2, 1, 2, 2].freeze,
+        [354, 2, 2, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2].freeze,
+        [354, 2, 2, 1, 2, 1, 2, 1, 2, 1, 1, 2, 1].freeze,
+        [384, 2, 2, 1, 2, 4, 2, 1, 2, 1, 2, 1, 1].freeze,
+        [355, 2, 1, 2, 2, 1, 2, 1, 2, 2, 1, 2, 1].freeze,
+        [355, 1, 2, 1, 2, 1, 2, 2, 1, 2, 2, 1, 2].freeze,
+        [384, 1, 4, 1, 2, 1, 2, 1, 2, 2, 1, 2, 2].freeze,
+        [354, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 2, 2].freeze,
+        [384, 2, 1, 2, 1, 1, 2, 3, 2, 1, 2, 2, 2].freeze,
+        [354, 2, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 2].freeze,
+        [354, 2, 1, 2, 1, 2, 1, 2, 1, 1, 2, 1, 2].freeze,
+        [384, 2, 1, 2, 2, 5, 1, 2, 1, 1, 2, 1, 2].freeze,
+        [354, 1, 2, 2, 1, 2, 2, 1, 2, 1, 1, 2, 1].freeze,
+        [355, 2, 1, 2, 1, 2, 2, 1, 2, 2, 1, 2, 1].freeze, # 2049
+      ].freeze
+
       MAX_YEAR_NUMBER = 150
 
-      # Given the region, CALENDAR_YEAR_INFO_MAP looks up the date 
+      # Given the region, CALENDAR_YEAR_INFO_MAP looks up the date
       # table and uses it in the calculation
       CALENDAR_YEAR_INFO_MAP = {
         kr: KOREAN_LUNAR_YEAR_INFO,
-        vi: VIETNAMESE_LUNAR_YEAR_INFO
+        vi: VIETNAMESE_LUNAR_YEAR_INFO,
+        hk: CHINESE_LUNAR_YEAR_INFO,
```

Comment:
I happen to see your PR. Could u also add taiwan region? so I can add Taiwan holidays later 😃
```suggestion
        hk: CHINESE_LUNAR_YEAR_INFO,
        tw: CHINESE_LUNAR_YEAR_INFO,
```

Reply:
I suggest that you add Taiwan in a separate PR along with the list of holidays for the sake of clarity.